platform/posix/ota_pal/source/ota_pal_posix.c

```diff
@@ -410,58 +410,66 @@ OtaPalStatus_t otaPal_Abort( OtaFileContext_t * const C )
     return OTA_PAL_COMBINE_ERR( mainErr, subErr );
 }
 
+
+
 OtaPalStatus_t otaPal_CreateFileForRx( OtaFileContext_t * const C )
 {
     OtaPalStatus_t result = OTA_PAL_COMBINE_ERR( OtaPalUninitialized, 0 );
     char realFilePath[ OTA_FILE_PATH_LENGTH_MAX ];
     OtaPalPathGenStatus_t status = OtaPalFileGenSuccess;
 
-    if( C != NULL )
+    assert( C != NULL );
+    assert( C->pFilePath != NULL );
+
+    if( C->pFilePath[ 0 ] != ( uint8_t ) '/' )
+    {
+        status = getFilePathFromCWD( realFilePath, ( const char * ) C->pFilePath );
+    }
+    else
     {
-        if( C->pFilePath != NULL )
+        ( void ) strncpy( realFilePath, ( const char * ) C->pFilePath, strlen( ( const char * ) C->pFilePath ) + 1U );
```

Comment:
A comment about the reason for `+ 1U` would be good to add.

Reply:
done